h5py/h5t.pyx

```diff
@@ -1994,3 +2004,19 @@ cpdef object py_get_vlen(object dt_in):
     warn("Deprecated; use check_vlen_dtype(dtype) instead",
         H5pyDeprecationWarning)
     return check_vlen_dtype(dt_in)
+
+def register_py_type(dtype dt_in):
```

Comment:
This should probably be `register_numpy_type`, since it specifically prepends "NUMPY:" in the tag. Also, I am happy for this to be public (so that user's can register their own custom numpy dtypes), but it will need a docstring :wink:

Reply:
Would it be nice to do automatic registration for users, make it transparent instead of throwing errors?

Reply:
Yeah, but that is probably beyond the scope of this PR.

Reply:
Would `register_dtype` be okay?